app/models/bookmark.rb

```diff
@@ -1,9 +1,15 @@
 # frozen_string_literal: true
 
 class Bookmark < ActiveRecord::Base
+  self.ignored_columns = [
+    "topic_id" # TODO (martin) (2021-12-01): remove
+  ]
+
   belongs_to :user
   belongs_to :post
-  belongs_to :topic
+
+  delegate :topic, to: :post
+  delegate :topic_id, to: :post
```

Comment:
Instead of delegation, perhaps an association might be clearer? `belongs_to :topic, through: :post`

Reply:
Oh I did mean to do that, I think I just updated the topic has_many through association but not this one. I would like to keep the `topic_id` delegation though, otherwise there are more plugins I have to update, and I don't see any reason not to use it.

Reply:
Ah yes the `topic_id` one is fine :+1: